### PR TITLE
Support `flexWrap` in `Box`

### PR DESCRIPTION
**Pull request by jodevsa**
adding support for flexWrap

## motivation:
Building a flashcard app
Screenshot 2021-10-03 at 01 35 35

Also contributes to #464

### CODE REVIEW

**Comment by jodevsa** [comment]
#478  must be merged first

**Comment by jodevsa** [comment]
@sindresorhus

File: src/styles.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ import {LiteralUnion} from 'type-fest';
 import {ForegroundColor} from 'chalk';
 
 export interface Styles {
+	readonly flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
```
Comment (sindresorhus) [inline src/styles.ts:8]: Needs doc comments, like the readme updates.

File: test/flex-wrap.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			<Text>C</Text>
+		</Box>
+	);
+
+	t.is(output, ' CA\n  B');
+});
+
+test('row - wrap content reverse', t => {
+	const output = renderToString(
+		<Box flexDirection="column" height={2} width={3} flexWrap="wrap-reverse">
```
Comment (SimenB) [inline test/flex-wrap.tsx:66]:
```suggestion
		<Box flexDirection="row" height={2} width={3} flexWrap="wrap-reverse">
```

File: src/styles.ts [hunk 3]
```diff
@@ -200,6 +202,20 @@ const applyFlexStyles = (node: YogaNode, style: Styles): void => {
 		);
 	}
 
+	if ('flexWrap' in style) {
+		if (style.flexWrap === 'nowrap') {
+			node.setFlexWrap(Yoga.WRAP_NO_WRAP);
+		}
+
+		if (style.flexWrap === 'wrap') {
```
Comment (SimenB) [inline src/styles.ts:210]: `else if` for these since only one can be true at a time?

**Comment by jodevsa** [comment]
@sindresorhus @SimenB  adjusted :)

**Comment by jodevsa** [comment]
@sindresorhus @SimenB 

any updates? I need this on NPM :)

File: test/flex-wrap.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+test('row - no wrap', t => {
+	const output = renderToString(
+		<Box flexDirection="row" width={2}>
+			<Text>A</Text>
+			<Text>BC</Text>
+		</Box>
+	);
+
+	t.is(output, 'BC\n');
```
[the remark this answers is not in the document]
Comment (jodevsa) [reply test/flex-wrap.tsx:14]: @vadimdemedes sorry to spam you, but can this be merged?

**Comment by jodevsa** [comment]
@vadimdemedes i see that you reopened the PR. what is the next step to get this baked in?

**Comment by matteodepalo** [comment]
@vadimdemedes @jodevsa would love to get this in! Could we reopen the PR and work towards merging it?

**Comment by jodevsa** [comment]
@matteodepalo  same here :( it seems like that this repo is not maintained anymore. I tried hard to push for someone merging my change.

**Comment by matteodepalo** [comment]
@jodevsa that's a shame. I think it's be worth keeping the PR open if you think it's complete and wait for a maintainer's comment. Is there any specific reason you closed it?

**Comment by jodevsa** [comment]
@matteodepalo not really, I've closed it to be able to track the things I'm working on as I usually ignore my notifications because they are distracting and stressful. I hope someday someone would improve GitHub's notification.



I've reopened the PR. @vadimdemedes it would be really nice if we can get this in someday :)

**Review by vadimdemedes — APPROVED** [review]
Thank you and sorry for taking so long 💛

**Pull request merged**